### ISSUE TITLE
sse is not auto reconnecting

### ISSUE THREAD

**Issue by aguegu** [tserkov/vue-sse#5]
Even the doc says it would, it does not. 

It get this message a lot: 'The connection to http://*******/sse was interrupted while the page was loading.' in firefox.

For reference: 
https://stackoverflow.com/questions/24564030/is-an-eventsource-sse-supposed-to-try-to-reconnect-indefinitely

**Comment by tserkov** [comment]
Like you've discovered, there are all kinds of reasons why EventSource might not connect -- it's up to the user agent implementation.

If it's the initial connect that fails, EventSource definitely won't retry.

If it's a lost connection once a previous connection was established, it should attempt to reconnect after a user-agent determined reconnection timeout.

I would recommend implementing an onError callback and manually triggering a reconnect, as depicted in the README:
```javascript
        // Catch any errors (ie. lost connections, etc.)
        sse.onError(e => {
          console.error('lost connection; giving up!', e);

          // This is purely for example; EventSource will automatically
          // attempt to reconnect indefinitely, with no action needed
          // on your part to resubscribe to events once (if) reconnected
          sse.close();
        });
```

**Comment by aguegu** [comment]
Yes, I got my reconnection mechanism setup and it works fine since then. I wish there could be some logging or event listener if sse is doing its auto-reconnection. Because from my point of view, it did not see any. 

A typical case is that ui is served by @vue/cli as a SPA with a proxy to an exprees-vue api. If I shut down the api service, I can see the sse connection broke. But if I turn the api service back on, I did not sse vue-sse got reconnected automatically.
